src/IO/ParallelReadBuffer.cpp

```diff
@@ -121,26 +137,25 @@ off_t ParallelReadBuffer::seek(off_t offset, int whence)
                 handleEmergencyStop();
 
             auto next_segment = front_worker->nextSegment();
-            if (static_cast<size_t>(offset) < current_position + next_segment.size())
+            current_position += next_segment.size();
+            if (offset < current_position)
             {
                 current_segment = std::move(next_segment);
                 working_buffer = internal_buffer = Buffer(current_segment.data(), current_segment.data() + current_segment.size());
-                current_position += current_segment.size();
                 pos = working_buffer.end() - (current_position - offset);
                 addReaders();
                 return offset;
             }
-
-            current_position += next_segment.size();
         }
+        chassert(false);
```

Comment:
Why?

Reply:
Just to make it obvious that this branch always returns, when skimming the code. But looks like it's more confusing than helpful, removing.